cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureDnsZoneService.java

```diff
@@ -0,0 +1,290 @@
+import com.sequenceiq.cloudbreak.cloud.azure.resource.AzureResourceIdProviderService;
+import com.sequenceiq.cloudbreak.cloud.azure.task.dnszone.AzureDnsZoneCreationCheckerContext;
+import com.sequenceiq.cloudbreak.cloud.azure.task.dnszone.AzureDnsZoneCreationPoller;
+import com.sequenceiq.cloudbreak.cloud.azure.view.AzureNetworkView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.notification.PersistenceNotifier;
+import com.sequenceiq.cloudbreak.cloud.notification.PersistenceRetriever;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.common.api.type.CommonStatus;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Service
+public class AzureDnsZoneService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureDnsZoneService.class);
+
+    private static final int DEPLOYMENT_LENGTH_LIMIT = 64;
+
+    private static final String DNS_ZONES = "-dns-zones";
+
+    private static final String NETWORK_LINKS = "-links";
+
+    @Inject
+    private AzureNetworkDnsZoneTemplateBuilder azureNetworkDnsZoneTemplateBuilder;
+
+    @Inject
+    private PersistenceRetriever resourcePersistenceRetriever;
+
+    @Inject
+    private PersistenceNotifier persistenceNotifier;
+
+    @Inject
+    private AzureDnsZoneCreationPoller azureDnsZoneCreationPoller;
+
+    @Inject
+    private AzureResourceIdProviderService azureResourceIdProviderService;
+
+    @Inject
+    private AzureUtils azureUtils;
+
+    @Value("${cb.arm.privateendpoint.services:}")
+    private List<String> privateEndpointServices;
+
+    public void getOrCreateDnsZones(AuthenticatedContext authenticatedContext, AzureClient azureClient, AzureNetworkView networkView,
+            String resourceGroup, Map<String, String> tags) {
+
+        String networkId = networkView.getNetworkId();
+        String networkResourceGroup = networkView.getResourceGroupName();
+        List<AzurePrivateDnsZoneServiceEnum> enabledPrivateEndpointServices = getEnabledPrivateEndpointServices();
+
+        boolean dnsZonesDeployed = azureClient.checkIfDnsZonesDeployed(resourceGroup, enabledPrivateEndpointServices);
+        boolean networkLinksDeployed = azureClient.checkIfNetworkLinksDeployed(resourceGroup, networkId, enabledPrivateEndpointServices);
+        String deploymentName = generateDeploymentName(enabledPrivateEndpointServices, DNS_ZONES);
+        String dnsZoneDeploymentId = azureResourceIdProviderService.generateDeploymentId(azureClient.getCurrentSubscription().subscriptionId(),
+                resourceGroup, deploymentName);
+
+        if (dnsZonesDeployed && networkLinksDeployed) {
+            LOGGER.debug("Dns zones ({}) and network links already deployed in resource group {}", enabledPrivateEndpointServices, resourceGroup);
```

Comment:
I don't think it's needed anymore

Reply:
It is needed to short-circuit the creation if it is not needed.